.github/workflows/release.yml

```diff
@@ -0,0 +1,20 @@
+name: PyPI Release
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  release:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Setup dependencies
+        uses: ./.github/actions/setup-dependencies
```

Comment:
I would just inline the setup-dependencies action here, as it is not reusable in this repo.